interpreter/cling/lib/Utils/PlatformPosix.cpp

```diff
@@ -56,24 +57,20 @@ namespace {
     }
 
   public:
-    PointerCheck() : FD(::open("/dev/random", O_WRONLY)) {
-      if (FD == -1) ::perror("open('/dev/random')");
-    }
-    ~PointerCheck() {
-      if (FD != -1) ::close(FD);
+    PointerCheck() {
+      page_size = ::sysconf(_SC_PAGESIZE);
     }
 
     bool operator () (const void* P) {
-      if (FD == -1)
-        return false;
       // std::find is considerably slower, do manual search instead.
       if (P == lines[0] || P == lines[1] || P == lines[2] || P == lines[3]
           || P == lines[4] || P == lines[5] || P == lines[6] || P == lines[7])
         return true;
 
       // There is a POSIX way of finding whether an address
       // can be accessed for reading.
-      if (::write(FD, P, 1/*byte*/) != 1) {
+      void *base = (void *)((((const size_t)P) / page_size) * page_size);
```

Comment:
Given that this is very sensitive for performance, I think that here it can be safely assumed that `page_size` is a power of 2, so that we can eliminate the division and replace it with a simpler expression, like the one below.
```
void *base = (((const size_t)P) & ~(page_size - 1));
```